homeassistant/components/sensor/luftdaten.py

```diff
@@ -95,48 +104,54 @@ def name(self):
     @property
     def state(self):
         """Return the state of the device."""
-        return self._state
+        try:
+            return self.luftdaten.data.values[self.sensor_type]
+        except AttributeError:
+            pass
 
     @property
     def unit_of_measurement(self):
         """Return the unit of measurement of this entity, if any."""
         return self._unit_of_measurement
 
-    def update(self):
-        """Get the latest data from REST API and update the state."""
-        self.rest_client.update()
-        value = self.rest_client.data
-
-        if value is None:
-            self._state = None
-        else:
-            parsed_json = json.loads(value)
-
-            log_entries_count = len(parsed_json) - 1
-            latest_log_entry = parsed_json[log_entries_count]
-            sensordata_values = latest_log_entry['sensordatavalues']
-            for sensordata_value in sensordata_values:
-                if sensordata_value['value_type'] == self.sensor_type:
-                    self._state = sensordata_value['value']
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        if self.luftdaten is None:
+            return
+
+        attr = {
+            ATTR_ATTRIBUTION: CONF_ATTRIBUTION,
+            ATTR_SENSOR_ID: self._sensor_id,
+            'lat': self.luftdaten.data.meta['latitude'],
+            'long': self.luftdaten.data.meta['longitude'],
+        }
+        return attr
+
+    @asyncio.coroutine
+    def async_update(self):
+        """Get the latest data from luftdaten.info and update the state."""
+        try:
+            yield from self.luftdaten.async_get()
```

Comment:
Wrong method call.

Reply:
What would the correct call be?

Reply:
Sorry I missed that you had changed the name, and I assumed the data class also would have a method called `update` or `async_update` since that is what we usually use.

Nevermind this comment. :blush:

Reply:
I will rename it to be consistent with the other platforms.